qsiprep/workflows/dwi/merge.py

```diff
@@ -115,7 +114,9 @@ def init_merge_and_denoise_wf(
             bids_dwi_files=raw_dwi_files,
             b0_threshold=config.workflow.b0_threshold,
             harmonize_b0_intensities=not config.workflow.no_b0_harmonization,
-            scan_metadata={scan: get_metadata_for_nifti(scan) for scan in raw_dwi_files},
+            scan_metadata={
+                scan: config.execution.layout.get_metadata(scan) for scan in raw_dwi_files
```

Comment:
much better!

Reply:
The only issue is that `read_nifti_sidecar` is still used in places. It looks like there are a few interfaces/functions (`get_distortion_grouping` and `get_best_b0_topup_inputs_from`) that use `read_nifti_sidecar` and do some complicated stuff, so I didn't try to remove it.